code/modules/antagonists/shadowling/shadowling_abilities.dm

```diff
@@ -0,0 +1,1095 @@
+
+	to_chat(user, "<span class='shadowling'>You dim the APC's screen and carefully begin siphoning its power into the void.</span>")
+	if(!do_after(user, 200, target=target_apc))
+		//Whoops!  The APC's light turns back on
+		to_chat(user, "<span class='shadowling'>Your concentration breaks and the APC suddenly repowers!</span>")
+		target_apc.set_light(2)
+		target_apc.visible_message("<span class='warning'>The [target_apc] begins glowing brightly!</span>")
+	else
+		//We did it
+		to_chat(user, "<span class='shadowling'>You return the APC's power to the void, disabling it.</span>")
+		target_apc.cell?.charge = 0	//Sent to the shadow realm
+		target_apc.chargemode = 0 //Won't recharge either until an engineer hits the button
+		target_apc.charging = 0
+		target_apc.update_icon()
+
+/obj/effect/proc_holder/spell/self/blindness_smoke //Spawns a cloud of smoke that blinds non-thralls/shadows and grants slight healing to shadowlings and their allies
+	name = "Blindness Smoke"
+	desc = "Spews a cloud of smoke which will blind enemies."
+	panel = "Shadowling Abilities"
+	charge_max = 600
+	human_req = TRUE
+	clothes_req = FALSE
+	action_icon_state = "black_smoke"
+	action_icon = 'icons/mob/actions.dmi'
+	sound = 'sound/effects/bamf.ogg'
+
+/obj/effect/proc_holder/spell/self/blindness_smoke/cast(mob/living/carbon/human/user) //Extremely hacky
+	if(!shadowling_check(user))
+		revert_cast()
+		return
+	user.visible_message("<span class='warning'>[user] bends over and coughs out a cloud of black smoke!</span>")
+	to_chat(user, "<span class='shadowling'>You regurgitate a vast cloud of blinding smoke.</span>")
+	var/obj/item/reagent_containers/glass/beaker/large/B = new /obj/item/reagent_containers/glass/beaker/large(user.loc) //hacky
+	B.reagents.clear_reagents() //Just in case!
+	B.invisibility = INFINITY //This ought to do the trick
+	B.reagents.add_reagent(/datum/reagent/shadowling_blindness_smoke, 10)
+	var/datum/effect_system/smoke_spread/chem/S = new
+	S.attach(B)
+	if(S)
+		S.set_up(B.reagents, 4, 0, B.loc)
+		S.start()
+	qdel(B)
+
+/obj/effect/proc_holder/spell/aoe_turf/unearthly_screech //Damages nearby windows, confuses nearby carbons, and outright stuns silly cones
+	name = "Sonic Screech"
+	desc = "Deafens, stuns, and confuses nearby people. Also shatters windows."
+	panel = "Shadowling Abilities"
+	range = 7
+	charge_max = 300
+	human_req = TRUE
+	clothes_req = FALSE
+	action_icon_state = "screech"
+	action_icon = 'icons/mob/actions.dmi'
+	sound = 'sound/effects/screech.ogg'
+
+/obj/effect/proc_holder/spell/aoe_turf/unearthly_screech/cast(list/targets,mob/user = usr)
+	if(!shadowling_check(user))
+		revert_cast()
+		return
+	user.audible_message("<span class='warning'><b>[user] lets out a horrible scream!</b></span>")
+	for(var/turf/T in targets)
+		for(var/mob/target in T.contents)
+			if(is_shadow_or_thrall(target))
+				if(target == user) //No message for the user, of course
+					continue
+				else
+					continue
```

Comment:
What ?

Reply:
Uhhh idk why did the original author make it, will fix